File: wasm_tests/test_2.js
// Same as last time

export function test_2() {
    function fibonacci(num) {
        if (num === 1) {
            return 1;
        } else if (num === 2) {
            return 1;
        } else return fibonacci(num - 1) + fibonacci(num - 2);
    }

    fibonacci(35);
}

File: wasm_tests/test_2.rs
// Explicitly use integers

pub fn test_2() {
    fn fibonacci(n: u32) -> u32 {
        match n {
            0_u32 => 1_u32,
            1_u32 => 1_u32,
            _ => fibonacci(n - 1_u32) + fibonacci(n - 2_u32),
        }
    }

    std::hint::black_box(fibonacci(35));
}
